synapse/config/_util.py

```diff
@@ -86,7 +86,7 @@ def parse_and_validate_mapping(
     try:
         # type-ignore: mypy doesn't like constructing `Dict[str, model_type]` because
         # `model_type` is a runtime variable. Pydantic is fine with this.
-        instances = parse_obj_as(Dict[str, model_type], config)  # type: ignore[valid-type]
+        instances = parse_obj_as(Dict[str, model_type], config)  # type: ignore
```

Comment:
Do we need to ignore multiple rules here? Should we instead use `# type: ignore[valid-type,something]`

Reply:
reivilibre fixed in [a5d500f](https://github.com/matrix-org/synapse/pull/15708/commits/a5d500fa46b6bc2062428f0d324b5802e0fc33d1)

(When I tried to do it, mypy didn't like me, thanks reivilibre!)